src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/InteropServices/ProvidePublicParameterlessSafeHandleConstructorTests.cs

```diff
@@ -0,0 +1,403 @@
+        {
+            string source = @"
+Class Foo
+    Private Sub New()
+    End Sub
+End Class";
+            await VerifyVB.VerifyAnalyzerAsync(source);
+        }
+
+        [Fact]
+        public async Task SafeHandleDerivedType_WithParameterlessConstructor_NoDiagnostics_CS()
+        {
+            string source = @"
+using Microsoft.Win32.SafeHandles;
+
+class FooHandle : SafeHandleZeroOrMinusOneIsInvalid
+{
+    public FooHandle() : base(true)
+    {
+    }
+
+    protected override bool ReleaseHandle() => true;
+}";
+            await VerifyCS.VerifyAnalyzerAsync(source);
+        }
+
+        [Fact]
+        public async Task SafeHandleDerivedType_WithParameterlessConstructor_NoDiagnostics_VB()
+        {
+            string source = @"
+Imports Microsoft.Win32.SafeHandles
+Public Class FooHandle : Inherits SafeHandleZeroOrMinusOneIsInvalid
+    Public Sub New()
+        MyBase.New(True)
+    End Sub
+    
+    Protected Overrides Function ReleaseHandle() As Boolean
+        Return True
+    End Function
+End Class";
+
+            await VerifyVB.VerifyAnalyzerAsync(source);
+        }
+
+        [Fact]
+        public async Task SafeHandleDerived_WithNonPublicParameterlessConstructor_CodeFix_CS()
+        {
+            string source = @"
+using Microsoft.Win32.SafeHandles;
+
+class FooHandle : SafeHandleZeroOrMinusOneIsInvalid
+{
+    private {|CA1418:FooHandle|}() : base(true)
+    {
+    }
+
+    protected override bool ReleaseHandle() => true;
+}";
+            string fixedSource = @"
+using Microsoft.Win32.SafeHandles;
+
+class FooHandle : SafeHandleZeroOrMinusOneIsInvalid
+{
+    public FooHandle() : base(true)
+    {
+    }
+
+    protected override bool ReleaseHandle() => true;
+}";
+
+            await VerifyCS.VerifyCodeFixAsync(source, fixedSource);
+        }
+
+        [Fact]
+        public async Task SafeHandleDerived_WithNonPublicParameterlessConstructor_CodeFix_VB()
+        {
+            string source = @"
+Imports Microsoft.Win32.SafeHandles
+Public Class FooHandle : Inherits SafeHandleZeroOrMinusOneIsInvalid
+
+    Private Sub {|CA1418:New|}()
+        MyBase.New(True)
+    End Sub
+    
+    Protected Overrides Function ReleaseHandle() As Boolean
+        Return True
+    End Function
+
+End Class";
+            string fixedSource = @"
+Imports Microsoft.Win32.SafeHandles
+Public Class FooHandle : Inherits SafeHandleZeroOrMinusOneIsInvalid
+
+    Public Sub New()
+        MyBase.New(True)
+    End Sub
+    
+    Protected Overrides Function ReleaseHandle() As Boolean
+        Return True
+    End Function
+
+End Class";
+
+            await VerifyVB.VerifyCodeFixAsync(source, fixedSource);
+        }
+
+        [Fact]
+        public async Task SafeHandleDerived_WithNoParameterlessConstructor_WithAccessibleBaseTypeParameterlessConstructor_CodeFix_CS()
+        {
+            string source = @"
+using System;
+using Microsoft.Win32.SafeHandles;
+
+abstract class FooHandle : SafeHandleZeroOrMinusOneIsInvalid
```

Comment:
In dotnet/roslyn, Usages of `Foo` is forbidden. Not sure if the same applies to dotnet/roslyn-analyzers. @mavasani.